### PR TITLE
Disable deleting the prometheus configmap on vmo startup

### PR DESCRIPTION
**Pull request by lake-of-dreams**
Prevents customizations done to prometheus configmap before vmo restart to be lost after any reboot.

### CODE REVIEW

**Comment by kminder** [comment]
This seems like something that there should be a test for.

**Comment by lake-of-dreams** [comment]
Test added in https://github.com/verrazzano/verrazzano/pull/2839

File: pkg/vmo/configmap.go [hunk 1]
```diff
@@ -15,6 +16,7 @@ import (
 	"github.com/verrazzano/verrazzano-monitoring-operator/pkg/constants"
 	"github.com/verrazzano/verrazzano-monitoring-operator/pkg/resources"
 	"github.com/verrazzano/verrazzano-monitoring-operator/pkg/resources/configmaps"
+	"gopkg.in/yaml.v2"
```
Comment (kminder) [inline pkg/vmo/configmap.go:19]: Do we have a PLS approval to use this?

File: pkg/vmo/configmap_test.go [hunk 2]
```diff
@@ -50,32 +49,6 @@ func TestCreateConfigmaps(t *testing.T) {
 	assert.Equal(t, 8, len(all.Items))
 }
 
-// TestDeleteConfigMapIfExists tests the deleteConfigMapIfExists function
```
Comment (kminder) [inline pkg/vmo/configmap_test.go:53]: This change deletes a test but doesn't add any tests.  There needs to be a unit test for reconcilePrometheusConfigMap

Comment (lake-of-dreams) [reply pkg/vmo/configmap_test.go:53]: Added tests

File: pkg/vmo/configmap.go (hunk 1 above)
Comment (lake-of-dreams) [reply pkg/vmo/configmap.go:20]: This [already](https://github.com/verrazzano/verrazzano-monitoring-operator/blob/0dfe30610f3c6b765fd1a990ab392751df72da46/pkg/config/config.go#L10) being used in this repo and is in TPL so I assume yes!

**Review by kminder — APPROVED** [review]
Approved.

**Pull request merged**